### PR TITLE
Remove RM99 caseless revolver

### PR DESCRIPTION
**Pull request by tenmillimaster**
### Summary

```SUMMARY: Content "Removes RM-99 caseless revolver"```
### Purpose of change
Rivtech firearms need to have a serious looking-at with respect to their feasibility. As these firearms should closely match real life analogues, the Rivtech RM99 revolver must be removed.

The RM-99 is unrealistic because...

- Modern development of advanced, prototype revolvers is essentially a non-starter. There is no market niche for this to be a commercial product, people will just buy a normal revolver. There is no military application, ordnance boards will simply utilize their conventional pistols, or other PDW-type weapons, or issue a rifle. Cutting edge development doesn't happen with the least important of the service weapons, the sidearm.
- All 8x40 is (or should be) telescoped, caseless ammunition, because that is what the nearest analogue, 4.73×33mm, is. Which means in a conventional firearm, there would be a significant amount of 'freebore' before the projectile engages the rifling, since a significant portion of the chamber area is now internal to the cartridge. This, on its own, is not an insurmountable engineering challenge. However, combined with the forcing cone and cylinder gap of a revolver, the projectile now has to travel a significant distance inside the telescoped case, then hit the forcing cone, and then engage the rifling. A similarly long internal freebore cartridge, the 7.62x38mmR nagant, expands into the forcing cone of the revolver, ensuring alignment.
image
image

 The RM-99 would need the cylinder to engage the forcing cone physically, like in the nagant revolver. Alternatively, the ammunition would need precisely concentric placement of the projectile to not have terrible accuracy, or cylinder shaving, a difficult manufacturing problem when working with a novel 'case' material such as plastic-bonded explosives. Both solutions are unacceptable complications for a niche-less design.  Chambers that do not correct for such freebore have accuracy issues.
- The inherent nature of telescoped, caseless ammunition means the cylinder would be quite large in relation to the diameter of the projectile, which isn't usually an issue for most revolvers. This would make the cylinder prohibitively large. 
image

- If the 8x40 cartridge is to be simply a scaled up 4.73×33mm caseless cartridge (which I say it must be, on the principle of nearest neighbor analogues), the revolver would have *square* cylinders. It'd look a little something like the  Dardick revolver, a dismal commercial failiure (but y'know, square instead of triangular). 
image

TLDR: Getting a novel ammo to work with a revolver is a waste of time.

#### Describe the solution
Migrates the RM99 to the RM103A.
Migrates the associated speedloader to the correct magazine.

#### Describe alternatives you've considered
None.
#### Testing
Spawned both items, pre-migration. Applied migration and reloaded save.

~~```src/savegame_json.cpp:2330 [void item::io(Archive&) [with Archive = io::JsonObjectInputArchive]] Item rm103a_pistol was loaded with charges, but can not have any!```~~

~~Results in above error, which I consider acceptable, but am open to input.~~

### CODE REVIEW

**Comment by jeremyshannon** [comment]
>Modern development of advanced, prototype revolvers is essentially a non-starter.

I would counter that the America of CDDA differs from our own. It appears that its populace is significantly more gun-happy and less risk-averse than our own, and their government and corporations seem rather enamored of the idea of novel and in some cases outlandish solutions to problems. A more 1950s outlook, with all the good and bad that comes with that.

**Comment by tenmillimaster** [comment]
>  I would counter that the America of CDDA differs from our own.

It does not.

**Comment by AlexanderBodnya** [comment]
While I understand the reasoning behind proposed changes, IMO a lot of cool content is being removed on basis it is not feasible. Wouldn't it be cool to shoot super sci-fi revolver? It certainly would be. Popularity of revolvers in FPS games is really high, everyone likes them. Certainly you can remove it on a basis that it is not feasible to produce such revolvers, but for me and I assume many others it would make game less cool.

**Comment by NoahGoldFox** [comment]
I agree with alexander. It may be slightly unrealistic, but it does make the game cooler and i dont think many people at all are actually complaining about it being in the game.

**Comment by Zireael07** [comment]
I don't think we had any complaints about the RM99, as opposed to coilguns and the like...

**Comment by FlameStormer2000** [comment]
> It does not.
It does in fact. 
https://cataclysmdda.org/lore-background.html
Rivtech is obviously one of the DARPA shell corporations. That's the why/how they had the resources to develop practical caseless ammo. The idea they would create a revolver using their propriety ammo to sell on the civilian market (just look at all the big game revolvers on the market today) even if it's just as a cover for funds acquired elsewhere is more then a good enough reason to keep it.

**Comment by tenmillimaster** [comment]
Range toys and big revolvers are just outgrowths of fundamentally sound, proven engineering, with tons of experience and iterative development to back it.

Our lore isn't "oh we just have better engineering", and our differences are intended to be small.


A caseless revolver is breaking new ground. 

Why would you waste so much time, for a small market, when you also have to justify your upscaled G11 is at least 200% better than an M4/m16? The profits just are not there. If it's a simple thing, like "oh ok, we'll just take a super hot cartridge and stuff it into a Browning tilting barrel action, and dump a heavy weight on the muzzle" e.g. the FK BRNO field pistol, or "revolver, but bigger"(BFR), or 'hey what if we Bastardize an AR and a normal pistol and stuff that into a big pistol', fine.  

But there are engineering hurdles that need to be overcome, and the action is so different from the G11 as to not be a simple extension of an idea, but something novel.

**Comment by jeremyshannon** [comment]
>The profits just are not there

Flamestormer's saying that potential profits are not that important if you're basically faking your sales numbers anyway, in order to launder the money you got by selling that neo-plutonium that you took from an alternate universe without notifying your governmental superiors.
 The alleged profits are guaranteed, you just need some products to push your money through, and if they're "revolutionary" products it makes your huge profits seem less suspicious.

**Comment by tenmillimaster** [comment]
That would require fudging Form 4473's and matching call ins to NICs, and
also a bunch of Interesting transactions with FFL's, and also spoofing tax
information. Basically inventing many people who don't exist.


Firearms are something the FBI (NICs) and the ATF checks in on. There are
better ways to launder money.

**Comment by tenmillimaster** [comment]
The migration no longer appears to result in a load error, following the pocket changes.
